src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -708,14 +709,15 @@ namespace AppInstaller::CLI::Workflow
         auto &source = context.Get<Execution::Data::Source>();
         bool shouldShowSource = source.IsComposite() && source.GetAvailableSources().size() > 1;
 
+        PinBehavior pinBehavior = context.Args.Contains(Execution::Args::Type::IncludePinned) ? PinBehavior::IncludePinned : PinBehavior::ConsiderPins;
         for (const auto& match : searchResult.Matches)
         {
             auto installedVersion = match.Package->GetInstalledVersion();
 
             if (installedVersion)
             {
-                auto latestVersion = match.Package->GetLatestAvailableVersion();
-                bool updateAvailable = match.Package->IsUpdateAvailable();
+                auto latestVersion = match.Package->GetLatestAvailableVersion(pinBehavior);
+                bool updateAvailable = match.Package->IsUpdateAvailable(pinBehavior);

```

Comment:
I think for `winget list`, we are hiding latest available versions from users forever if a pin is set, maybe we should just apply the pin behavior for upgrade related scenarios

Reply:
I think we need to have a mechanism to show if a package is pinned in list so a user could see when new versions are available. The pinning might be temporary to avoid a known bug, and the user might still be looking for any new versions.

I believe the same is true for winget upgrade. We might want a separate section for pinned packages so we can still show if new versions are available for the same reason.

Reply:
I added a count of user-pinned packages at the end of the update list, and I'm now showing the latest available version on list even if it is pinned.